crates/ruff_db/Cargo.toml

```diff
@@ -13,7 +13,7 @@ license = { workspace = true }
 [dependencies]
 ruff_annotate_snippets = { workspace = true }
 ruff_cache = { workspace = true, optional = true }
-ruff_diagnostics = { workspace = true }
+ruff_diagnostics = { workspace = true, features = ["serde"] }
```

Comment:
With the approach you've taken now, the correct gating here is to make the `ruff_diagnostic/serde` feature dependent on whether the `serde` feature is enabled for `ruff_db` (similar to what we do with camino)